ocs_ci/ocs/benchmark_operator.py

```diff
@@ -0,0 +1,215 @@
+BMO_NAME = "benchmark-operator"
+# The benchmark operator git repository
+BMO_REPO = "https://github.com/cloud-bulldozer/benchmark-operator"
+# The label used by the benchmark operator for the cache drop pods
+BMO_LABEL = "kernel-cache-dropper"
+# The benchmark operator deployment resource
+BMO_DEPLOYMENT = "deployment/benchmark-controller-manager"
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope="function")
+def benchmark_operator(request):
+    """
+    Use the benchmark operator as fixture
+
+    """
+
+    def teardown():
+        operator.cleanup()
+
+    request.addfinalizer(teardown)
+    operator = BenchmarkOperator()
+    operator.deploy()
+    return operator
+
+
+class BenchmarkOperator(object):
+    """
+    Workload operation using Benchmark-Operator
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Initializer function.
+
+        Initialize object variables, clone the benchmark operator repo.
+        and label the worker nodes.
+
+        Args:
+            kwargs (dict):
+                Following kwargs are valid
+                repo: benchmark-operator repo to used - a github link
+                branch: branch to use from the repo
+
+        Example Usage:
+            r1 = BenchmarkOperator()
+            r1.deploy()
+            # use oc apply to apply custom modified bench
+            my_custom_bench = my_custom_bench.yaml
+            run_cmd('oc apply -f my_custom_bench')
+
+        """
+        log.info("Initialize the benchmark-operator object")
+        self.args = kwargs
+        self.repo = self.args.get("repo", BMO_REPO)
+        self.branch = self.args.get("branch", "master")
+        # the namespace is a constant for the benchmark-operator
+        self.namespace = BMO_NAME
+        self.pgsql_is_setup = False
+        self.ocp = OCP()
+        self.ns_obj = OCP(kind="namespace")
+        self.pod_obj = OCP(namespace=BMO_NAME, kind="pod")
+        # list of worker nodes to label
+        self.worker_nodes = get_worker_nodes()
+        self._clone_operator()
+        self.dir += f"/{BMO_NAME}"
+
+        # to use the cache dropping pod, worker nodes need to be labeled.
+        log.info("Labeling the worker nodes for cache-dropping enable.")
+        try:
+            helpers.label_worker_node(
+                self.worker_nodes, label_key=BMO_LABEL, label_value="yes"
+            )
+        except CommandFailed:
+            # this is probably because of the nodes are already labeled, so,
+            # checking if nodes are labeled and continue anyway.
+            result = self.pod_obj.exec_oc_cmd(f"get node -l {BMO_LABEL}")
+            found = [node for node in self.worker_nodes if re.search(node, result)]
+            if len(found) == len(self.worker_nodes):
+                log.info("All worker nodes are labeled")
+            else:
+                log.warning("Labeling nodes failed, Not all workers node are labeled !")
```

Comment:
is it worth continuing the test if all worker nodes are not labeled? I believe we might not get consistent results.

Reply:
since the cache dropping is not implemented in all benchmarks used with the benchmark operator (e.g pgbench) we do not want to failed the test because of this.
this is in responsibility of the test and not of the benchmark-operator object.